wikipedia_searcher.py

```diff
@@ -39,7 +40,7 @@ def __init__(self, **kwargs):
                 # The options property contains a list of titles of Wikipedia pages that the query may refer to.
                 self.may_refer = e.options
                 # Removing duplicates in lists.
-                self.may_refer = list(set(self.may_refer))
+                self.may_refer = list(set(self.may_refer))[:self.titles]
```

Comment:
what the behavior if not set?

Reply:
it defaults to None thanks to line 23 `self.titles = kwargs.get('titles', None)` and serves back all the results